temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1580,6 +1600,22 @@ private void processContinueAsNewWorkflowExecution(
         parentChildInitiatedEventId);
   }
 
+  private void processWorkflowCompletionCallbacks(RequestContext ctx) {
```

Comment:
Can you please add tests that confirm this is triggered for every terminal workflow state and that the callbacks will be carried over CaN and workflow retries?

Reply:
I did add the tests for terminal states and CaN. The test server doesn't support workflow retries AFAICT, so I skipped that. I also skipped an explicit test for cron runs since it uses the same code path as CaN but I can always add something if you think it's necessary.